plugins/woo-ai/src/image-background-removal/image_utils/uploadImageToLibrary.ts

```diff
@@ -0,0 +1,75 @@
+type BackgroundRemovalParams = {
+	imageBlob: Blob;
+	libraryFilename: string;
+};
+
+declare global {
+	interface Window {
+		ajaxurl?: string;
+	}
+}
+
+export const uploadImageToLibrary = async ( {
+	imageBlob,
+	libraryFilename,
+}: BackgroundRemovalParams ) => {
+	const fileObj = new File( [ imageBlob ], libraryFilename );
+	let _fileId: number | null = null;
+
+	await wp.mediaUtils.uploadMedia( {
+		allowedTypes: 'image',
+		filesList: [ fileObj ],
+		onError: ( e: Error ) => {
+			throw e;
+		},
+		onFileChange: ( files: Array< { id: number } > ) => {
+			if ( files.length > 0 && files[ 0 ]?.id ) {
+				_fileId = files[ 0 ].id;
+			}
+		},
+	} );
+
+	if ( _fileId === null ) {
+		return;
+	}
+
+	const nonceValue = window?.JP_CONNECTION_INITIAL_STATE.apiNonce ?? '';
+	const ajaxUrl = window?.ajaxurl;
+
+	if ( ! ( ajaxUrl && nonceValue ) ) {
+		throw new Error( 'Missing nonce or ajaxurl' );
+	}
+
+	const formData = new FormData();
+
+	formData.append( 'action', 'get-attachment' );
+	formData.append( 'id', String( _fileId ) );
+	formData.append( '_ajax_nonce', nonceValue );
+
+	const response = await fetch( ajaxUrl, {
+		method: 'POST',
+		body: formData,
+	} ).then( ( res ) => res.json() );
+
+	if ( ! response.data ) {
+		return;
+	}
+
+	const fileData = {
```

Comment:
Can we only create a variable for the attachment and skip creating the fileData? I think only the attachment is being used in the code since we can get the file ID from the `_fileId` variable.

Reply:
Sounds good, done in 08205dfef0c71ae839fbaa574d557ba5290f8500